crates/graphql-federated-graph/src/federated_graph.rs

```diff
@@ -51,10 +52,37 @@ pub struct FederatedGraph {
 }
 
 impl FederatedGraph {
+    #[cfg(all(feature = "from_sdl", feature = "extension"))]
```

Comment:
Do we want to feature flag extensions?

Reply:
We want to feature flag IO in the graphql-federated-graph crate, because we release that on crates.io. It's the only open source GraphQL composition crate AFAIK, so it would be great to keep it IO free for non-grafbase, or non-extensions users.